src/image.rs

```diff
@@ -267,9 +267,8 @@ pub trait ImageDecoder: Sized {
             {
                 let from = &tmp[x as usize * bpp..width as usize * bpp];
 
-                let to = &mut buf[i * width as usize * bpp..width as usize * bpp];
-
-                ::copy_memory(from, to);
+                let start = i * width as usize;
+                buf[start..start + from.len()].copy_from_slice(from);
```

Comment:
It looks like you're trying to fix a bug here at the same time, but it still looks wrong to me. I haven't tested, but I would expect it to be:
```rust
let from = &tmp[x as usize * bpp..][..width as usize * bpp];
let to = &mut buf[i * width as usize * bpp..][..width as usize * bpp];
to.copy_from_slice(from);
```

Reply:
Your code looks correct.
Thanks.